rfcs/0000-multiple-control-plane-providers.md

```diff
@@ -0,0 +1,270 @@
+
+While option 1 above implies a lazy configuration of the functional components into the gateway providers, option 2 perhaps could be implemented in a rather proactive fashion, such as the one proposed in this RFC.
+
+By linking the `Kuadrant` CR directly to the gateways to be configured, maybe the Operator could infer the provider kinds and provider control plane configuration objects behind those gateways and thus proceed with the injection of the functional components at this level. However, this would be an approach yet to be proven. Without guarantees that the [`GatewayClass`](https://gateway-api.sigs.k8s.io/api-types/gatewayclass/) objects referenced by the [`Gateway`](https://gateway-api.sigs.k8s.io/api-types/gateway/) instances point unequivocally to the gateway provider kind and configuration object in a way that the Operator knows how to inject the configuration assertively, this may or may not be a feasible option after all.
+
+As an example of the problem of detecting the exact kind of gateway provider and implementing the configuration accordingly, without it being based again on convention or heuristic, consider the lack of distiction between Istio and OpenShift Service Mesh (OSSM) regarding the structure of their defined `GatewayClass` objects. In fact, such distinction perceived at the level of a `Gateway` object may even go against the purpose of abstraction embodied in the definition of the `GatewayClass` kind, as stated in Gateway API's [specification](https://gateway-api.sigs.k8s.io/api-types/gatewayclass/#gatewayclass):
+
+> We expect that one or more `GatewayClasses` will be created by the infrastructure provider for the user. It allows decoupling of which mechanism (e.g. controller) implements the `Gateways` from the user. For instance, an infrastructure provider may create two `GatewayClasses` named `internet` and `private` to reflect `Gateways` that define Internet-facing vs private, internal applications.
+>
+> [...]
+>
+> The user of the classes will not need to know how `internet` and `private` are implemented.
+
+Differences between `GatewayClasses` provided by different implementations such as the [GatewayClass parameters](https://gateway-api.sigs.k8s.io/api-types/gatewayclass/#gatewayclass-parameters) and [GatewayClass controller selection](https://gateway-api.sigs.k8s.io/api-types/gatewayclass/#gatewayclass-controller-selection) are, according to the spec, only "encouraged" or "RECOMMENDED".
+
+# Prior art
+[prior-art]: #prior-art
+
+N/A.
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+1. Is having multiple service mesh control planes (i.e. multiple gateway providers) within the same cluster a real use case or rather just a technical possibility that we choose not to support?<br/>
```

Comment:
it seems it is a real scenerio  based on recent conversations where there may be two control planes one for the gateway and one for the mesh